### DISCUSSION TITLE
No functions under workspace

### DISCUSSION THREAD

**Discussion by janeliu650** [open-webui/open-webui#7304] [General]
After installed latest verion 0.4.4, there is no functions menu in workspace. There is no way to assign a function to model.
After I imported fucntions, when clicking the editing icon(pencil), trying to pickup fucntion  it says:

To select filters here, add them to the "Functions" workspace first.

But there is no functions under workspace.

**Comment by tjbck** [comment]
Please read the changelogs.

<img width="903" alt="image" src="https://github.com/user-attachments/assets/b227c0cb-cab6-4f66-946d-ca14bf8afd62">

**Comment by janeliu650** [comment]
Then how to assign a function to a model?

**Reply by silentoplayz** [reply]
The same way you could before; Within the model's edit page itself. You can enable tools, filters, and action buttons for either base or custom models directly within Open WebUI.

e.g. image
![image](https://github.com/user-attachments/assets/dcf00447-92c3-4f87-8ea6-eeef8fbc8d5a)

**Comment by mifas** [comment]
I'm using the latest version, as suggested in the doc, I have enabled the function in Admin, and nothing is showing in the custom model section. I'm very new to this, please guide me on this thanks
![image](https://github.com/user-attachments/assets/4a3a63e4-9e4f-4e8c-9848-962e55862d41)
![image](https://github.com/user-attachments/assets/29b63956-d48f-4929-aae3-fe44c24588dc)

**Reply by silentoplayz** [reply]
Pipe/Manifold models (when said pipe or manifold is toggled on) will not be displayed within the Custom Models section, as `Functions` are an Admin toggle-able feature. They can be found in the `Models` section of the Admin settings instead.

**Reply by mifas** [reply]
> They can be found in the Models section of the Admin settings instead.

I'm unable to find it under the Admin settings. Could you please elaborate further?
![image](https://github.com/user-attachments/assets/b2488e80-efc3-4157-baed-eed88ebf4298)

**Reply by silentoplayz** [reply]
When a `Pipe` function is toggled on, any models it provides will be available in the `Models` section for modification in the Admin Settings.
![image](https://github.com/user-attachments/assets/95363cbd-a95e-4fc6-b54d-b74471c0ac0c)
![image](https://github.com/user-attachments/assets/088b2eea-6426-4c40-bfd2-25c2bbb595e5)

**Comment by jaipaddy** [comment]
After importing the Assistant function for OpenAI from https://openwebui.com/f/mantonherre/assistant_openai/, how do I make my gpt-4o-mini model recognize this function? I see "openai/Custom Assistant" as a model option in the `Models` section in Admin `Settings` with the correct `assistant_openai.asst_*` id that I created within OpenAI and its vector storage. On the chat UI, when I choose this model and ask for relevant data from the files uploaded, it shows 4 highlighted lines and no text with references to the 2 files I uploaded.

**Comment by sparksbenjamin** [comment]
Have we found a solution for this.  I understand it has been moved to Admin setting and is now tied to the models but I am still not setting the feature to enable the function.  
![image](https://github.com/user-attachments/assets/9caa249d-e5ca-49df-8c63-167b8107daf6)
![image](https://github.com/user-attachments/assets/070836f9-5f3f-4fa9-810d-f533e7857a8c)

**Reply by silentoplayz** [reply]
As I've mentioned before (https://github.com/open-webui/open-webui/discussions/7304#discussioncomment-11426308 & https://github.com/open-webui/open-webui/discussions/7304#discussioncomment-11448392), only filters and action button functions will appear for selection on a model's edit page. A pipe function works differently. I see no issue here @sparksbenjamin. See [Open WebUI documentation on Functions](https://docs.openwebui.com/features/plugin/functions/) for reference.

**Reply by isCopyman** [reply]
> As I've mentioned before ([#7304 (reply in thread)](https://github.com/open-webui/open-webui/discussions/7304#discussioncomment-11426308) & [#7304 (reply in thread)](https://github.com/open-webui/open-webui/discussions/7304#discussioncomment-11448392)), only filters and action button functions will appear for selection on a model's edit page. A pipe function works differently. I see no issue here @sparksbenjamin. See [Open WebUI documentation on Functions](https://docs.openwebui.com/features/plugin/functions/) for reference.

I'm not sure if I'm understanding correctly that these pipe functions are equivalent to forming a base model?Then in the workspace, when I build the model, the base model I select corresponds to the model of this pipe function?

**Reply by silentoplayz** [reply]
You can select a Pipe function as the base model for custom models in the Workspace section.
